core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1829,7 +1897,49 @@ private[spark] class BlockManager(
     data.dispose()
   }
 
+  /**
+   * Class to handle block manager decommissioning retries
+   * It creates a Thread to retry offloading all RDD cache blocks
+   */
+  private class BlockManagerDecommissionManager {
+    @volatile private var stopped = false
+    private val cacheReplicationThread = new Thread {
```

Comment:
Same comment about the thread up above, I'm not sure why we need to be in a separate thread (but maybe I've missed something).

Reply:
Tried to explain why we have created a thread here - https://github.com/apache/spark/pull/27864#issuecomment-605599508. Basically this BlockManager will keep on retrying offloading of cache blocks every 30 seconds.

Reply:
Makes sense. I guess we can just (later on) send a message back once all the blocks are done migrating.

Reply:
When all the blocks are done migrating - Driver will automatically know. BlockManagerMaster is continuously receiving BlockUpdates. So when all blocks are offloaded - Dynamic allocation will atomatically know that this executor doesn't have cache data and If other criteria are met (like executorIdleTimeout etc), then the executor will automatically be removed by ExecutorMonitor class of DynamicAllocation.